src/main/java/de/dm/infrastructure/logcapture/ExpectedKeyValueLogstashDelegate.java

```diff
@@ -0,0 +1,36 @@
+package de.dm.infrastructure.logcapture;
+
+import net.logstash.logback.marker.SingleFieldAppendingMarker;
+
+import java.util.Arrays;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+import static java.lang.String.format;
+import static java.lang.System.lineSeparator;
+
+// this is a delegate accessing net.logstash so that ExpectedKeyValue can fail gracefully
```

Comment:
Why did you remove logstash in the build but than verify that it exists?

Reply:
Logstash is a build-time dependency so it can be used for assertions **if** it is in the classpath.

However, it should not be a dependency. This is why I check at runtime if it is present.

This is also why this package-private delegate exists: It will never be loaded if logstash ist not in the classpath to avoid ClassNotFoundException